File: matlab_src/aitken_iteration.m
function result = aitken_iteration(g, x_0, tolerance)
    k = 1;
    x_current = x_0;
    x_1 = g(x_current);
    x_2 = g(x_1);
    x_next = x_2 - (x_2 - x_1)^2 / (x_2 - 2 * x_1 + x_current);
    fprintf('%d result: %.14f\n', k, x_next);

    while abs(x_next - x_current) >= tolerance
        x_current = x_next;
        x_1 = g(x_current);
        x_2 = g(x_1);
        x_next = x_2 - (x_2 - x_1)^2 / (x_2 - 2 * x_1 + x_current);
        result = x_next;
        k = k + 1;
        fprintf('%d result: %.14f\n', k, result);
    end
end

File: python_src/aitken_iteration.py
import numpy as np

def aitken_iteration(iterative, x_0, epsilon):
    k = 1
    x_cur = x_0
    
    while True:
        x_1 = iterative(x_cur)
        x_2 = iterative(x_1)
        x_next = x_2 - (x_2 - x_1)**2 / (x_2 - 2 * x_1 + x_cur)
        
        if abs(x_next - x_cur) < epsilon:
            print(f"{k} iteration: {x_next}")
            break
        
        x_cur = x_next
        print(f"{k} iteration: {x_cur}")
        k += 1
        
def iterative(x):
    return (x + 1)**(1 / 3)

aitken_iteration(iterative, 1, 1e-8)
        